packages/myst-templates/src/template.ts

```diff
@@ -65,8 +65,15 @@ class MystTemplate {
     if (!fs.existsSync(templateYmlPath)) {
       throw new Error(`The template yml at "${templateYmlPath}" does not exist`);
     }
+
+    interface TemplateYaml {
+      files: string[];
+      [key: string]: any;
+    };
     const content = fs.readFileSync(templateYmlPath).toString();
-    return yaml.load(content);
+    const yamlData : TemplateYaml = yaml.load(content) as TemplateYaml;
+    const { files, ...filteredYaml } = yamlData;
+    return filteredYaml;
```

Comment:
The `files` are used primarily in the tex/typst templates to copy assets like styles, images, etc. over into a new build folder. So removing files here will break that.

I am assuming that the `package-lock.json` is complaining (not sure on your local setup)? We could pass a flag, like `ignoreTemplateFiles`, into the MystTemplate class and do this logic if true. e.g. if the `opts?.template` is passed into `getSiteTemplate` above.

Reply:
Yes, no `page-lock.json`, no `build` dir, etc.

Reply:
Added a flag `validateFiles` in the appropriate places, which seems to do the trick.

Reply:
I think just doing this logic when the template is specified makes sense, left a different comment on that!